### PR TITLE
Arm64 build fixes

### PR DESCRIPTION
**Pull request by gkhanna79**
@dotnet-bot test arm64 windows_nt

### CODE REVIEW

**Comment by gkhanna79** [comment]
@dotnet-bot test arm64 windows_nt

**Comment by gkhanna79** [comment]
@dotnet-bot test arm64 windows_nt

**Comment by gkhanna79** [comment]
@dotnet-bot test arm64 windows_nt

**Comment by gkhanna79** [comment]
@jkotas PTAL

File: src/gc/env/gcenv.base.h [hunk 1]
```diff
@@ -128,7 +128,17 @@ typedef DWORD (WINAPI *PTHREAD_START_ROUTINE)(void* lpThreadParameter);
...
+ 
+ #elif defined(_ARM64_)
+
+  extern "C" void __yield(void);
+  #pragma intrinsic(__yield)
+  __forceinline void YieldProcessor() { __yield();}
+
+  extern "C" void __dmb(const unsigned __int32 _Type);
+  #pragma intrinsic(__dmb)
+  #define MemoryBarrier() { __dmb(_ARM_BARRIER_SY); }
```
Comment (jkotas) [inline src/gc/env/gcenv.base.h]: This should be `__dmb(_ARM64_BARRIER_SY)`

**Comment by jkotas** [comment]
While you are on it, could you please make `/homeparams` to be AMD64-only in the root CMakeLists.txt. It will fix thousands of `Command line warning D9002: ignoring unknown option '/homeparams' ` warnings on ARM64.

**Comment by jkotas** [comment]
LGTM otherwise

**Comment by gkhanna79** [comment]
@dotnet-bot test arm64 windows_nt

**Pull request merged**